examples/ecs/run_criteria.rs

```diff
@@ -0,0 +1,51 @@
+//! Demonstrates how and why to use run criteria to control whether or not systems run
+
+/*
+    Basically, run criteria is the way to make a function run under certaion conditons.
```

Comment:
This paragraph could use a bit more clarity, and a more formal tone :)

I would try to cover:

1. What a run criteria is / what it does
2. Some practical uses
3. Any restrictions on what types of functions can be used as run criteria

Reply:
Alright